cabal-install/Distribution/Client/CmdConfigure.hs

```diff
@@ -83,9 +85,18 @@ configureAction (configFlags, configExFlags, installFlags, haddockFlags)
     baseCtx <- establishProjectBaseContext verbosity cliConfig
 
     -- Write out the @cabal.project.local@ so it gets picked up by the
-    -- planning phase.
-    writeProjectLocalExtraConfig (distDirLayout baseCtx)
-                                 cliConfig
+    -- planning phase. If old config exists, then print the contents
+    -- before overwriting
+    exists <- doesFileExist "cabal.project.local"
+    if exists
+      then do
+        contents <- readFile "cabal.project.local"
+        putStr contents
+        writeProjectLocalExtraConfig (distDirLayout baseCtx)
+                                      cliConfig
+      else
+        writeProjectLocalExtraConfig (distDirLayout baseCtx)
```

Comment:
Move these out of the conditional so that they are not duplicated. Once you do this, you can use 'when' above to avoid the conditional with empty branch.